pytorch/training/docker/1.12/py3/cu113/Dockerfile.trcomp.gpu

```diff
@@ -0,0 +1,125 @@
+
+########
+###   :::Update NCCL:::
+###   This is a temporary workaround to
+###   upgrade NCCL
+########
+
+# Remove Older NCCL
+RUN rm -rvf `find /usr/local | grep nccl` /usr/local/obj
+
+ENV NCCL_VERSION=2.12.12
+
+# Install Newer NCCL
+RUN cd /tmp \
+ && git clone https://github.com/NVIDIA/nccl.git -b v${NCCL_VERSION}-1 \
+ && cd nccl \
+ && make -j `nproc` src.build BUILDDIR=/usr/local \
+ && rm -rf /tmp/nccl
+
+# Reinstall AWS OFI NCCL
+
+ENV BRANCH_OFI=1.4.0-aws
+
+RUN mkdir /tmp/efa-ofi-nccl  && cd /tmp/efa-ofi-nccl  && git clone https://github.com/aws/aws-ofi-nccl.git -b v${BRANCH_OFI}  && cd aws-ofi-nccl  && ./autogen.sh  && ./configure --with-libfabric=/opt/amazon/efa   --with-mpi=/opt/amazon/openmpi   --with-cuda=/usr/local/cuda   --with-nccl=/usr/local --prefix=/usr/local  && make  && make install  && rm -rf /tmp/efa-ofi-nccl  && rm -rf /var/lib/apt/lists/*  && apt-get clean
+
+########
+###   :::Update NCCL:::
+########
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# Version args - overwritten by args specified in buildspec
+ARG PYTHON=python3
+
+ARG PT_BUCKET=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/trcomp/r1.12.0_sm/20221110-064048/f77b79fa779a512c319d20e42f59bbd3559bd16a
+ARG PT_URL=${PT_BUCKET}/torch-1.12.0%2Bcu113-cp38-cp38-linux_x86_64.whl
+ARG PT_XLA_URL=${PT_BUCKET}/torch_xla-1.12-cp38-cp38-linux_x86_64.whl
+ARG TORCHVISION_URL=${PT_BUCKET}/torchvision-0.13.0a0%2Bda3794e-cp38-cp38-linux_x86_64.whl
+ARG TORCHAUDIO_URL=${PT_BUCKET}/torchaudio-0.12.0%2B2e13884-cp38-cp38-linux_x86_64.whl
+ARG SMD_DATA_PARALLEL_URL=https://smdataparallel.s3.amazonaws.com/binary/pytorch/1.12.0/cu113-compiler/2022-08-09/smdistributed_dataparallel-1.5.0-cp38-cp38-linux_x86_64.whl
+
+
+# Install Sagemaker PythonSDK
+RUN pip install --no-cache-dir sagemaker
+
+
+RUN apt-get update \
+ # TODO: Remove upgrade statements once packages are updated in base image
+ && apt-get -y upgrade --only-upgrade systemd openssl cryptsetup \
+ && apt install -y git-lfs \
+ && apt install -y libomp5 \
+ && apt-get clean \  
+ && rm -rf /var/lib/apt/lists/*
+
+
+# Install PyTorch
+RUN pip uninstall -y torch \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${PT_URL} \
+ && rm -rf ${PT_URL}
+
+# Install PyTorch XLA
+RUN pip uninstall -y torch_xla \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${PT_XLA_URL} \
+ && rm -rf ${PT_XLA_URL}
+
+# Install TorchVision
+RUN pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${TORCHVISION_URL} \
+ && rm -rf ${TORCHVISION_URL}
+
+# Install TorchAudio
+RUN pip uninstall -y torchaudio \
+ && pip install --no-deps --no-cache-dir -U --force-reinstall ${TORCHAUDIO_URL} \
+ && rm -rf ${TORCHAUDIO_URL}
+
+ARG CONDA_PREFIX=/opt/conda
+
+# Numpy version gets downgraded and becomes incompatible with PyTorch wheel in the above conda command.
+RUN pip install numpy --upgrade
```

Comment:
lets remove this, numpy is already installed using conda. See #2416 for the context on incompatibility

Reply:
which one?